### ISSUE TITLE
Implement TTIRToTTNN pass to capture a ttnn.logaddexp OP

### ISSUE THREAD

**Issue by mrakitaTT** [tenstorrent/tt-mlir#690]
In case of training, if JAX code is written in a [way to be numerically stable and avoid softmax computation](https://gregorygundersen.com/blog/2020/02/09/log-sum-exp/) by directly calculating log likelihood (logits - logsumexp), then logsumexp is used. StableHLO doesn't have Logsumexp OP and implements it as a combination of bunch of simpler OPs ([constant](https://openxla.org/stablehlo/spec#constant), [maximum](https://openxla.org/stablehlo/spec#maximum), [exponential](https://openxla.org/stablehlo/spec#exponential), [reduce](https://openxla.org/stablehlo/spec#reduce), [divide](https://openxla.org/stablehlo/spec#divide), [broadcast_in_dim](https://openxla.org/stablehlo/spec#broadcast_in_dim)). We need to add TTNN Logaddexp OP which lowers to `metal::ttnn::logaddexp` function. We then need to detect this pattern of converted TTIR ops and match it with TTNN.Logaddexp.

**Comment by staylorTT** [comment]
@mrakitaTT Do we have an associated TTNN issue, seems like this needs TTNN support?

**Comment by mrakitaTT** [comment]
This was blocked on implementing fusion pass in tt-mlir, which recently happened so we can now attend to this task. This is a perf improvement so I am not sure about priority, I've now raised attention to this and couple other similar tasks [here](https://github.com/tenstorrent/tt-mlir/issues/688#issuecomment-2891570568).

>seems like this needs TTNN support?

No we just need to implement this op in our TTIR/TTNN dialects, support in metal ttnn already exists with `ttnn::logaddexp` op.
